cosmo_tester/test_suites/test_security/secured_nodecellar_test.py

```diff
@@ -0,0 +1,113 @@
+        self.test_manager_blueprint_path = path(mb_path)
+        self.test_inputs_path = path(inputs_path)
+
+        self._update_manager_blueprint_security()
+        self._bootstrap()
+        self._set_credentials_env_vars()
+        self._running_env_setup()
+
+        self._test_openstack_nodecellar('openstack-blueprint.yaml')
+
+    def _copy_manager_blueprint(self):
+        return util.generate_unique_configurations(
+            workdir=self.workdir,
+            original_inputs_path=self.env.cloudify_config_path,
+            original_manager_blueprint_path=self.env._manager_blueprint_path)
+
+    def _update_manager_blueprint_security(self):
+        security_path = 'node_templates.manager.properties.cloudify.security'
+        security_settings = {
+            'enabled': 'true',
+            'userstore_driver': {
+                'implementation':
+                    'flask_securest.userstores.file:FileUserstore',
+                'properties': {
+                    'userstore_file': 'users.yaml',
+                    'identifying_attribute': 'username'
+                }
+            },
+            'authentication_methods': [
+                {
+                    'name': 'password',
+                    'implementation': 'flask_securest.'
+                                      'authentication_providers.password:'
+                                      'PasswordAuthenticator',
+                    'properties': {
+                        'password_hash': 'plaintext'
+                    }
+                },
+                {
+                    'name': 'token',
+                    'implementation': 'flask_securest.'
+                                      'authentication_providers.token:'
+                                      'TokenAuthenticator',
+                    'properties': {
+                        'secret_key': 'yaml_secret'
+                    }
+                }
+            ]
+        }
+        self._update_manager_blueprint(security_path, security_settings)
+
+    def _update_manager_blueprint(self, prop_path, new_value):
+        with util.YamlPatcher(self.test_manager_blueprint_path) as patch:
+            patch.set_value(prop_path, new_value)
+
+    def _bootstrap(self):
+        self.cfy.bootstrap(blueprint_path=self.test_manager_blueprint_path,
+                           inputs_file=self.test_inputs_path,
+                           task_retries=5,
+                           install_plugins=False)
+        self.addCleanup(self.cfy.teardown)
+
+    def _set_credentials_env_vars(self):
+        os.environ[CLOUDIFY_USERNAME_ENV] = TEST_CFY_USERNAME
+        os.environ[CLOUDIFY_PASSWORD_ENV] = TEST_CFY_PASSWORD
+
+    def _running_env_setup(self):
+        cfy = CfyHelper(cfy_workdir=self.workdir)
+        self.env.management_ip = cfy.get_management_ip()
+        self.client = CloudifyClient(
+            self.env.management_ip,
+            user=os.environ[CLOUDIFY_USERNAME_ENV],
```

Comment:
If the username/password constants are already available is see no need in reading the from the evironment as they were set only one function before. The constants can be used directly